modular_skyrat/modules/modular_implants/code/nifsoft_catalog.dm

```diff
@@ -0,0 +1,101 @@
+GLOBAL_LIST_INIT(available_nifsofts, list(
+	/datum/nifsoft/hivemind,
+	/datum/nifsoft/summoner,
+	/datum/nifsoft/shapeshifter,
+))
+
+/datum/computer_file/program/nifsoft_downloader
+	filename = "nifsoftcatalog"
+	filedesc = "NIFSoft Catalog"
+	extended_desc = "A virtual storefront that allows the user to install NIFSofts and purchase various NIF related products"
+	category = PROGRAM_CATEGORY_MISC
+	size = 3
+	tgui_id = "NtosNifsoftCatalog"
+	program_icon = "bag-shopping"
+	usage_flags = PROGRAM_TABLET
+	///What bank account is money being drawn out of?
+	var/datum/bank_account/paying_account
+	///What NIF are the NIFSofts being sent to?
+	var/datum/weakref/target_nif
+
+/datum/computer_file/program/nifsoft_downloader/Destroy(force)
+	. = ..()
+	paying_account = null
+	target_nif = null
+
+//TGUI STUFF
+
+/datum/computer_file/program/nifsoft_downloader/ui_data(mob/user)
+	var/list/data = list()
+
+	paying_account = computer.computer_id_slot?.registered_account || null
```

Comment:
compress 31 & 32

Reply:
I'm not going to do this because it is saving to a variable that is used at a later point in the code.